Tests/BraveWalletTests/AssetDetailStoreTests.swift

```diff
@@ -0,0 +1,510 @@
+    assetDetailBitcoinException.expectedFulfillmentCount = 11
+    store.$isBuySupported
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertFalse($0)
+      }
+      .store(in: &cancellables)
+    store.$isSendSupported
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertFalse($0)
+      }
+      .store(in: &cancellables)
+    store.$isSwapSupported
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertFalse($0)
+      }
+      .store(in: &cancellables)
+    store.$btcRatio
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertEqual($0, "1 BTC")
+      }
+      .store(in: &cancellables)
+    store.$priceHistory
+      .dropFirst()
+      .sink { priceHistory in
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertEqual(priceHistory.count, 1)
+        XCTAssertEqual(priceHistory[0].price, "0.99")
+      }
+      .store(in: &cancellables)
+    store.$price
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertEqual($0, "$28,324.00")
+      }
+      .store(in: &cancellables)
+    store.$priceIsDown
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertFalse($0)
+      }
+      .store(in: &cancellables)
+    store.$priceDelta
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertEqual($0, "0.58%")
+      }
+      .store(in: &cancellables)
+    store.$isLoadingPrice
+      .dropFirst()
+      .collect(2)
+      .sink { values in
+        defer { assetDetailBitcoinException.fulfill() }
+        guard let value = values.last
+        else {
+          XCTFail("Unexpected isLoadingPrice")
+          return
+        }
+        XCTAssertFalse(value)
+      }
+      .store(in: &cancellables)
+    store.$isInitialState
+      .dropFirst()
+      .sink {
+        defer { assetDetailBitcoinException.fulfill() }
+        XCTAssertFalse($0)
+      }
+      .store(in: &cancellables)
+    store.$isLoadingChart
+      .dropFirst()
+      .collect(2)
+      .sink { values in
+        defer {
+          XCTAssertNil(store.network)
+          XCTAssertTrue(store.accounts.isEmpty)
+          XCTAssertTrue(store.transactionSummaries.isEmpty)
+
+          assetDetailBitcoinException.fulfill()
+        }
+        guard let value = values.last
+        else {
+          XCTFail("Unexpected isLoadingChart")
+          return
+        }
+        XCTAssertFalse(value)
+      }
+      .store(in: &cancellables)
+    
+    store.update()
+    wait(for: [assetDetailBitcoinException], timeout: 1)
+    cancellables.removeAll()
+    
+    // setup store
+    store = AssetDetailStore(
```

Comment:
Awesome including test for two different tokens (one with buy supported, the other without). I think better to move this to a separate test function (one for Coin Market Bitcoin, one for Coin Market Ethereum)